java/number-formatting.md

```diff
@@ -0,0 +1,36 @@
+# NumberFormatting 
+
+## Assignment
+Format a number for a specific locale. For example, the number 12345.123 should be formatted as `12,345.12` in the USA locale, but `12.345,12` in the German locale.
+
+## Solution
+```java
+DecimalFormat currencyFormatter = (DecimalFormat) NumberFormat.getInstance(Locale.getDefault());
+
+currencyFormatter.setMaximumFractionDigits(2);
+currencyFormatter.setMinimumFractionDigits(2);
+
+System.out.println(currencyFormatter.format(123456.12));
+```
+
+Depending on your default Locale the currencyFormatter will format the number in the right way for you.
+
+##### If you have a locale like `Locale.US` it will output:
+```java
+-12,345.12
```

Comment:
To clarify, the code example in the markdown uses a positive number. The linked code example does use negatives.